modules/openapi-generator/src/main/resources/elm/Api.mustache

```diff
@@ -55,13 +56,18 @@ send toMsg req =
 
 
 sendWithCustomError : (Http.Error -> e) -> (Result e a -> msg) -> Request a -> Cmd msg
-sendWithCustomError mapError toMsg (Request req) =
+sendWithCustomError mapError toMsg req =
+    sendWithCustomExpect (expectJson mapError toMsg) req
+
+
+sendWithCustomExpect : (Json.Decode.Decoder a -> Http.Expect msg) -> Request a -> Cmd msg
+sendWithCustomExpect expect (Request req) =
```

Comment:
Not sure I'm following why this takes a decoder to message function. Can you give me a basic example of what you would to with it?

Reply:
The decoders in the signature are the ones provided by the `Request`s from `Api/Request/Default.elm`. To use `sendWithCustomExpect` one has to provide the logic to interpret the response body given the decoders.

My goal was make it possible to use a custom error type (the current `send` and `sendWithCustomError` both expect a `Http.Error`). A shortened example for `String`  that e.g. accesses the body of a `Http.BadStatus_` could e.g. look like this:

```elm
customSend : (Result String a -> msg) -> Api.Request a -> Cmd msg
customSend toMsg req =
    Api.sendWithCustomExpect (Http.expectStringResponse toMsg << decodeResponse) req


decodeResponse : Json.Decode.Decoder a -> Http.Response String -> Result String a
decodeResponse decoder response =
    case response of
        ...
        Http.BadStatus_ _ body ->
            Err body

        ...
```

Where `decodeResponse` goes from:

```elm
Json.Decode.Decoder a -> (Http.Response String -> Result String a)
```

and `expectStringResponse toMsg` from:

```elm
(Response String -> Result x a) -> Expect msg
```

Reply:
Ah yes, that makes sense to support. Thanks for clarifying!